## Transformação do dataset tarbase_data.csv
Semana 17/09 - 24/09
- Selecionar apenas as espécies de interesse
- Adicionar uma coluna com as sequencias de cada mirna

In [ ]:
# Import
import pandas as pd
import numpy as np

In [ ]:
df = pd.read_csv('/content/drive/My Drive/TCC - 2020/tarbase_data.csv', sep='\t')

In [ ]:
df.shape

(649378, 12)

In [ ]:
df.head()

geneId       geneName  ... up_down condition
0  0910001A06Rik  0910001A06Rik  ...      UP       NaN
1  1200004M23Rik  1200004M23Rik  ...    DOWN       NaN
2  1700027J05Rik  1700027J05Rik  ...      UP       NaN
3  1810015A11Rik  1810015A11Rik  ...    DOWN       NaN
4  2310047A01Rik  2310047A01Rik  ...    DOWN       NaN

[5 rows x 12 columns]

In [ ]:
# Filtrando as 5 espécies observadas
df = df.loc[df['species'].isin(['Arabidopsis thaliana','Oryza sativa','Zea mays','Medicago truncatula','Vitis vinifera'])]

In [ ]:
df.shape

(2660, 12)

In [ ]:
df.head()

geneId            geneName  ... up_down condition
51  AC233070_21.1(mtr)  AC233070_21.1(mtr)  ...    DOWN       NaN
52  AC233070_21.1(mtr)  AC233070_21.1(mtr)  ...    DOWN       NaN
53  AC233100_46.1(mtr)  AC233100_46.1(mtr)  ...    DOWN       NaN
54  AC233680_24.1(mtr)  AC233680_24.1(mtr)  ...    DOWN       NaN
55   AC233685_3.1(mtr)   AC233685_3.1(mtr)  ...    DOWN       NaN

[5 rows x 12 columns]

In [ ]:
# Eliminando algumas colunas
df = df.drop(columns=['category','method','positive_negative','direct_indirect','up_down','condition','cell_line','tissue'])
df.head()

geneId            geneName           mirna              species
51  AC233070_21.1(mtr)  AC233070_21.1(mtr)  mtr-miR396a-5p  Medicago truncatula
52  AC233070_21.1(mtr)  AC233070_21.1(mtr)  mtr-miR396b-5p  Medicago truncatula
53  AC233100_46.1(mtr)  AC233100_46.1(mtr)  mtr-miR1507-3p  Medicago truncatula
54  AC233680_24.1(mtr)  AC233680_24.1(mtr)    mtr-miR2643a  Medicago truncatula
55   AC233685_3.1(mtr)   AC233685_3.1(mtr)     mtr-miR159a  Medicago truncatula

In [ ]:
df.shape

(2660, 4)

In [ ]:
# Verificando quantos miRNA  únicos existem (desconsidera valores repetidos)
df['mirna'].nunique()

316

In [ ]:
# Verificando a frequência de cada miRNA
df['mirna'].value_counts()

## Arquivos fasta (.fa)



In [ ]:
!pip install biopython

In [ ]:
from Bio import SeqIO

lista_seq = []
lista_id = []

for fasta in SeqIO.parse('/content/drive/My Drive/TCC - 2020/mature.fa','fasta'):
    
    lista_id.append(fasta.id)
    lista_seq.append(str(fasta.seq))

In [ ]:
len(lista_id)

48885

In [ ]:
len(lista_seq)

48885

In [ ]:
print(lista_seq)

['UGAGGUAGUAGGUUGUAUAGUU', 'CUAUGCAAUUUUCUACCUUACC', 'UCCCUGAGACCUCAAGUGUGA', 'ACACCUGGGCUCUCCGGGUACC', 'CAUACUUCCUUACAUGCCCAUA', 'UGGAAUGUAAAGAAGUAUGUA', 'CAUCAAAGCGGUGGUUGAUGUG', 'UAUCACAGCCAGCUUUGAUGUGC', 'AGGCAGUGUGGUUAGCUGGUUG', 'ACGGCUACCUUCACUGCCACCC', 'UGCUGGUUUCUUCCACAGUGGUA', 'UCACCGGGUGGAAACUAGCAGU', 'CGCCAAUUUUCGCUUCAGUGCUA', 'UCACCGGGUGAAAAUUCGCAUG', 'UGUGGGUGUCCGUUGCGGUGCUA', 'UCACCGGGUGAACACUUGCAGU', 'UCCGGUUUUUUCCGUGGUGAUA', 'UCACCGGGAGAAAAACUGGAGU', 'AGCUGAUUUCGUCUUGGUAAUA', 'UCACCGGGUGUAAAUCAGCUUG', 'AGUGGAUGUAUGCCAUGAUGAUA', 'UCACCGGGUGUACAUCAGCUAA', 'GGUGGUUUUUCUCUGCAGUGAUA', 'UCACCGGGUGAAAAAUCACCUA', 'GUGGGUGUUUGCUUUUUCGGUGAAG', 'UCACCGGGUUAACAUCUACAGA', 'GACAUCAAGAAACUAGUGAUUAUG', 'UAUCACAGUUUACUUGCUGUCGC', 'CUGGAUGUGCUCGUUGGUCAUA', 'UGACUAGAGACACAUUCAGCU', 'CUGGAUGUGCUCGUUAGUCAUA', 'UGACUAGAGACACAUUCAGCU', 'AAGAGAGCCGUCUAUUGACAGU', 'UGUCAUGGAGUCGCUCUCUUCA', 'AAGAGAGCAGUCUAUUGACAGU', 'UGUCAUGGAGGCGCUCUCUUCA', 'UGAGGUAGGCUCAGUAGAUGCGA', 'ACAUCCACCAGCCUAGCUCGCA', 'C

In [ ]:
# Criando um dicionário a partir das listas
my_dict = {'id':lista_id,'sequencia':lista_seq}

In [ ]:
print(my_dict)

{'id': ['cel-let-7-5p', 'cel-let-7-3p', 'cel-lin-4-5p', 'cel-lin-4-3p', 'cel-miR-1-5p', 'cel-miR-1-3p', 'cel-miR-2-5p', 'cel-miR-2-3p', 'cel-miR-34-5p', 'cel-miR-34-3p', 'cel-miR-35-5p', 'cel-miR-35-3p', 'cel-miR-36-5p', 'cel-miR-36-3p', 'cel-miR-37-5p', 'cel-miR-37-3p', 'cel-miR-38-5p', 'cel-miR-38-3p', 'cel-miR-39-5p', 'cel-miR-39-3p', 'cel-miR-40-5p', 'cel-miR-40-3p', 'cel-miR-41-5p', 'cel-miR-41-3p', 'cel-miR-42-5p', 'cel-miR-42-3p', 'cel-miR-43-5p', 'cel-miR-43-3p', 'cel-miR-44-5p', 'cel-miR-44-3p', 'cel-miR-45-5p', 'cel-miR-45-3p', 'cel-miR-46-5p', 'cel-miR-46-3p', 'cel-miR-47-5p', 'cel-miR-47-3p', 'cel-miR-48-5p', 'cel-miR-48-3p', 'cel-miR-49-5p', 'cel-miR-49-3p', 'cel-miR-50-5p', 'cel-miR-50-3p', 'cel-miR-51-5p', 'cel-miR-51-3p', 'cel-miR-52-5p', 'cel-miR-52-3p', 'cel-miR-53-5p', 'cel-miR-53-3p', 'cel-miR-54-5p', 'cel-miR-54-3p', 'cel-miR-55-5p', 'cel-miR-55-3p', 'cel-miR-56-5p', 'cel-miR-56-3p', 'cel-miR-57-5p', 'cel-miR-57-3p', 'cel-miR-58a-5p', 'cel-miR-58a-3p', 'cel-miR-59-

In [ ]:
# Criar um dataframe a partir do dicionário
df_seq = pd.DataFrame(data=my_dict)

In [ ]:
df_seq.head()

id               sequencia
0  cel-let-7-5p  UGAGGUAGUAGGUUGUAUAGUU
1  cel-let-7-3p  CUAUGCAAUUUUCUACCUUACC
2  cel-lin-4-5p   UCCCUGAGACCUCAAGUGUGA
3  cel-lin-4-3p  ACACCUGGGCUCUCCGGGUACC
4  cel-miR-1-5p  CAUACUUCCUUACAUGCCCAUA

## Fazendo um merge (inner join) entre o dataframe com os genesId e o dataframe com as sequências

In [ ]:
new_df = pd.merge(df, df_seq,left_on='mirna', right_on='id', how='left')

In [ ]:
new_df.head()

geneId  ...               sequencia
0  AC233070_21.1(mtr)  ...   UUCCACAGCUUUCUUGAACUU
1  AC233070_21.1(mtr)  ...   UUCCACAGCUUUCUUGAACUG
2  AC233100_46.1(mtr)  ...  CCUCGUUCCAUACAUCAUCUAG
3  AC233680_24.1(mtr)  ...   UUUGGGAUCAGAAAUUAGAGA
4   AC233685_3.1(mtr)  ...   UUUGGAUUGAAGGGAGCUCUA

[5 rows x 6 columns]

In [ ]:
# Verificando valores ausentes no dataset
new_df.isnull().sum()

geneId       0
geneName     0
mirna        0
species      0
id           0
sequencia    0
dtype: int64

In [ ]:
new_df['sequencia'].describe()

count                     2660
unique                     177
top       UGACAGAAGAGAGUGAGCAC
freq                       252
Name: sequencia, dtype: object

In [ ]:
new_df['mirna'].describe()

count               2660
unique               316
top       zma-miR164d-5p
freq                  37
Name: mirna, dtype: object

In [ ]:
new_df['id'].describe()

count               2660
unique               316
top       zma-miR164d-5p
freq                  37
Name: id, dtype: object

**Podemos observar a partir dos valores unicos descritos acima que existem 316 mirna's unicos, porém só 177 sequências unicas. Logo, 139 sequências estão associadas a mais de um mirna.**

In [ ]:
# Verificando a quantidade de sequencia em cada mirna
pd.set_option("display.max_rows", 400)
new_df.groupby(['species','mirna'])['sequencia'].count()

In [ ]:
new_df.describe()

geneId       geneName  ...              id             sequencia
count            2660           2660  ...            2660                  2660
unique            417            417  ...             316                   177
top     GRMZM2G165488  GRMZM2G165488  ...  zma-miR164d-5p  UGACAGAAGAGAGUGAGCAC
freq               68             68  ...              37                   252

[4 rows x 6 columns]

In [ ]:
new_df.loc[new_df['mirna'] == 'ath-miR173-5p']

geneId   geneName  ...             id               sequencia
12  At1g50055  At1g50055  ...  ath-miR173-5p  UUCGCUUGCAGAGAGAAAUCAC
28  AT2G27400  AT2G27400  ...  ath-miR173-5p  UUCGCUUGCAGAGAGAAAUCAC
30  At2g39675  At2g39675  ...  ath-miR173-5p  UUCGCUUGCAGAGAGAAAUCAC
31  AT2G39681  AT2G39681  ...  ath-miR173-5p  UUCGCUUGCAGAGAGAAAUCAC
45  AT5G10950  AT5G10950  ...  ath-miR173-5p  UUCGCUUGCAGAGAGAAAUCAC

[5 rows x 6 columns]

In [ ]:
# Verificando a quantidade de especies em cada sequencia
pd.set_option("display.max_rows", 400)
new_df.groupby(['sequencia','mirna'])['species'].count()

In [181]:
del new_df['id']

In [182]:
new_df.head()

geneId  ...               sequencia
0  AC233070_21.1(mtr)  ...   UUCCACAGCUUUCUUGAACUU
1  AC233070_21.1(mtr)  ...   UUCCACAGCUUUCUUGAACUG
2  AC233100_46.1(mtr)  ...  CCUCGUUCCAUACAUCAUCUAG
3  AC233680_24.1(mtr)  ...   UUUGGGAUCAGAAAUUAGAGA
4   AC233685_3.1(mtr)  ...   UUUGGAUUGAAGGGAGCUCUA

[5 rows x 5 columns]

In [183]:
new_df.to_csv('/content/drive/My Drive/TCC - 2020/dataset_with_mirnaSeq.csv')